src/compiler/checker.ts

```diff
@@ -2609,7 +2609,11 @@ namespace ts {
                 if (!inTypeAlias && type.aliasSymbol && isTypeSymbolAccessible(type.aliasSymbol, context.enclosingDeclaration)) {
                     const name = symbolToTypeReferenceName(type.aliasSymbol);
                     const typeArgumentNodes = mapToTypeNodes(type.aliasTypeArguments, context);
-                    return createTypeReferenceNode(name, typeArgumentNodes);
+                    const ref = createTypeReferenceNode(name, typeArgumentNodes);
+                    if (type.aliasKind === AliasKind.KeyOf) {
+                        return createTypeOperatorNode(ref);

```

Comment:
a "TypeOperatorNode" is really a KeyofNode, isn't it? Would you be OK with changing the name? Doesn't have to be in this PR, just asking.

Reply:
`unique` should also be a type operator node - the factory was just never updated to account for that, I think, since it's always just made as `unique symbol` (so you'll never need anything else right now). So... the name should _probably_ be changed 🤷‍♂️